cmd/krel/cmd/gcbmgr_generate.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/spf13/cobra"
+
+	"k8s.io/release/pkg/gcp/build"
+)
+
+type GenerateOptions struct {
+	DateFrom       string
+	DateTo         string
+	DateFromParsed string
+	DateToParsed   string
+}
+
+var (
+	generateOpts = &GenerateOptions{}
+)
+
+// generateCmd is a krel gcbmgr subcommand which generate information about the command that the operator ran for a specific release cut
+var generateCmd = &cobra.Command{
+	Use:           "generate --branch release-1.18 --date-from 2020-06-18 [--date-to 2020-06-19]",
+	Short:         "Run generate to build a list of commands that ran when cutting a specific Kubernetes release",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return RunGenerate(generateOpts)
+	},
+}
+
+func init() {
+	generateCmd.PersistentFlags().StringVar(
+		&generateOpts.DateFrom,
+		"date-from",
+		"",
+		"Get the jobs starting from a specific date. Format to use yyyy-mm-dd",
+	)
+
+	generateCmd.PersistentFlags().StringVar(
+		&generateOpts.DateTo,
+		"date-to",
+		"",
+		"Get the jobs ending from a specific date. Format to use yyyy-mm-dd",
+	)
+
+	gcbmgrCmd.AddCommand(generateCmd)
+}
+
+// RunGenerate is the function invoked by 'krel gcbmgr generate', responsible for
+// getting the jobs and builind the list of commands to be added in the GitHub issue
+func RunGenerate(opts *GenerateOptions) error {
+	if err := opts.Validate(); err != nil {
+		return errors.Wrap(err, "validating generate options")
+	}
+
+	logrus.Infof("Running generate with the following options: %+v", opts)
+	logrus.Infof("Build options: %v", *buildOpts)
+
+	tagFilter := fmt.Sprintf("tags=%q create_time>%q create_time<%q", gcbmgrOpts.Branch, generateOpts.DateFromParsed, generateOpts.DateToParsed)
+	jobs, err := build.GetJobsByTag(buildOpts.Project, tagFilter)
+	if err != nil {
+		return errors.Wrap(err, "getting the GCP build jobs")
+	}
+
+	tableString := &strings.Builder{}
+	table := tablewriter.NewWriter(tableString)
+	table.SetAutoWrapText(false)
+
+	table.SetHeader([]string{"Step", "Command", "Link", "Start", "Duration", "Succeeded?"})
+	for i := len(jobs) - 1; i >= 0; i-- {
+		job := jobs[i]
+		stage := ""
+		for _, stageTag := range job.Tags {
+			if stageTag == "RELEASE" {
+				stage = stageTag
+				break
+			}
+			if stageTag == "STAGE" {
+				stage = stageTag
+				break
+			}
+		}
+
+		// Build the command that was executed
+		command := fmt.Sprintf("`krel gcbmgr --%s --type %s --branch %s %s",
+			strings.ToLower(stage),
+			job.Substitutions["_TYPE"],
+			job.Substitutions["_RELEASE_BRANCH"],
+			job.Substitutions["_BUILDVERSION"],
+		)
+
+		var mock string
+		if job.Substitutions["_NOMOCK"] != "" {
+			command = fmt.Sprintf("%s %s`", command, job.Substitutions["_NOMOCK"])
+			mock = ""
+		} else {
+			command = fmt.Sprintf("%s`", command)
+			mock = "MOCK "
+		}
+
+		start := job.Timing["BUILD"].StartTime
+		end := job.Timing["BUILD"].EndTime
+		status := job.Status
```

Comment:
Can we map that to just output `Yes` or `No`?

Reply:
done